app/models/document.rb

```diff
@@ -230,10 +230,7 @@ def publish
 
   def unpublish
     handle_remote_error do
-      Services.publishing_api.unpublish(content_id, type: 'gone')
-
-      AttachmentDeleteWorker.perform_async(content_id)
-      RummagerDeleteWorker.perform_async(base_path)
+      DocumentUnpublisher.unpublish(content_id, base_path)
```

Comment:
The idea was that the class name explains what the service is responsible for (`UnpublishDocument`), and `call` makes it easier to call these services without having to check what the method names are (`FindAllDocuments.call`, `PublishDocument.call`, etc.

But I'm happy with both solutions!